docs: document compatibility with other otel packages (#293)
embrace-io/embrace-web-sdk: README.md

## Troubleshooting

### Compatibility with OTel packages

The SDK is built on top of OpenTelemetry and as such it is possible to use it alongside other OTel libraries. If you
wish to customize the sdk behaviour by configuring custom resources, exporters, processors or instrumentations you

> should make sure to use the same versions of the OTel packages as the SDK uses. The next table summarizes the compatible

Reviewer: ```suggestion
should make sure to use versions of the OTel packages that are compatible with what the SDK uses. The next table summarizes those compatible
```

versions of the OTel packages:

| Open Telemetry APIs | Core  | Instrumentations & Contrib |
|---------------------|-------|----------------------------|

> | ^1.9.0              | ^1.30 | ^0.57.0                    |

Reviewer 2: The instrumentation one is actually tricky. We saw that `"@opentelemetry/instrumentation-document-load": "0.45.0",` depends on 0.200 but 0.44.1 does not.. Should we add a separate column for that one?

Author: I believe the only weird one is instrumentation-document-load, I am not even sure why that is not in sync with every other instrumentation. I will add a note, but I doubt anyone will face the issue, as if they want that specific instrumentation, we already provide it 🙂

Reviewer: hmm if it's the only one with that issue it might be fine to leave off, since we include that instrumentation there shouldn't be a reason for the user to as well (even if they included a compatible one they'd end up with duplicated telemetry)

Author: the only scenario I can think of is if they disable our instrumentation because they want their own to further customize it.

Author: I agree with Jon, I will probably remove the note, it is too specific

Reviewer 2: hehe, sounds good!
